src/services/tray.js

```diff
@@ -51,43 +51,47 @@ const registerVisibilityListener = () => {
 }
 
 const updateTrayMenu = () => {
-    const mainWindow = windowService.getMainWindow();
+    if (process.platform != 'darwin') {
+        // Menus under macos cause multiple changeVisibility triggers 
+        const mainWindow = windowService.getMainWindow();
 
-    const contextMenu = Menu.buildFromTemplate([
-        {
-            label: isVisible ? 'Hide' : 'Show',
-            type: 'normal',
-            click: () => {
-                if (isVisible) {
-                    mainWindow.hide();
-                } else {
-                    mainWindow.show();
+        const contextMenu = Menu.buildFromTemplate([
+            {
+                label: isVisible ? 'Hide' : 'Show',
+                type: 'normal',
+                click: () => {
+                    if (isVisible) {
+                        mainWindow.hide();
+                    } else {
+                        mainWindow.show();
+                        mainWindow.focus();
+                    }
+                }
+            },
+            {
+                type: 'separator'
+            },
+            {
+                label: 'Quit',
+                type: 'normal',
+                click: () => {
+                    mainWindow.close();
                 }
-            }
-        },
-        {
-            type: 'separator'
-        },
-        {
-            label: 'Quit',
-            type: 'normal',
-            click: () => {
-                mainWindow.close();
-            }
-        },
-    ]);
+            },
+        ]);
```

Comment:
There seem to be many whitespace changes which are unnecessary and obscure the real change..